blockchain/types/tests/digest.rs

```diff
@@ -212,55 +190,65 @@ fn origin_block_digestible_ast() {
     digestible_test_case_ast("test", &origin, expected_ast);
 }
 
-#[test]
-fn block_contents_digestible_test_vectors() {
-    let results = test_blockchain(BlockVersion::ONE);
+fn test_blockchain(block_version: BlockVersion) -> Vec<[u8; 32]> {
+    let origin = Block::new_origin_block(&test_origin_tx_outs());
 
-    // Test digest of block contents
-    assert_eq!(
-        results[0].1.hash().0,
-        [
-            141, 3, 218, 9, 245, 129, 12, 99, 192, 12, 107, 216, 4, 191, 254, 93, 247, 103, 110,
-            204, 77, 126, 154, 187, 198, 139, 58, 24, 191, 179, 59, 125
-        ]
-    );
-    assert_eq!(
-        results[1].1.hash().0,
-        [
-            254, 230, 153, 200, 129, 98, 221, 154, 120, 76, 165, 230, 183, 128, 60, 26, 202, 64,
-            171, 237, 164, 209, 9, 170, 109, 54, 133, 143, 110, 215, 199, 69
-        ]
-    );
-    assert_eq!(
-        results[2].1.hash().0,
-        [
-            215, 120, 30, 77, 126, 121, 230, 151, 225, 205, 247, 66, 101, 113, 221, 32, 122, 41,
-            29, 53, 98, 48, 72, 183, 219, 255, 228, 161, 47, 46, 245, 221
-        ]
-    );
+    let recipients = test_accounts()
+        .iter()
+        .map(|account| account.default_subaddress())
+        .collect::<Vec<_>>();
 
-    let results = test_blockchain(BlockVersion::ZERO);
+    let mut rng = FixedRng::from_seed([10u8; 32]);
+    get_blocks_with_recipients(block_version, 3, &recipients, 1, 5, 50, origin, &mut rng)
+        .into_iter()
+        .map(|block_data| {
+            let hash = &block_data.block().contents_hash;
+            // Sanity check
+            assert_eq!(hash, &block_data.contents().hash());
+            hash.0
+        })
+        .collect()
+}
 
-    // Test digest of block contents
-    assert_eq!(
-        results[0].1.hash().0,
-        [
-            127, 243, 87, 10, 229, 29, 137, 126, 255, 243, 163, 147, 39, 214, 116, 139, 171, 229,
-            167, 113, 37, 54, 91, 191, 40, 244, 194, 114, 234, 189, 209, 239
-        ]
-    );
+// Test digest of block contents at versions 0 and 1.
+#[test]
+fn block_contents_digestible_v0() {
     assert_eq!(
-        results[1].1.hash().0,
+        test_blockchain(BlockVersion::ZERO),
         [
-            62, 59, 155, 219, 6, 106, 54, 48, 148, 22, 117, 39, 251, 161, 22, 50, 158, 7, 5, 104,
-            182, 182, 7, 2, 156, 24, 0, 200, 68, 170, 76, 242
+            [
+                203, 245, 7, 219, 1, 210, 70, 210, 93, 109, 135, 251, 188, 10, 120, 155, 108, 240,
```

Comment:
Because I changed how the blocks are generated, and I couldn't find a way to make it compatible with the old blocks generations. So the options here are to change the bytes, or copy the old code that generated blocks here, which seems unwieldy.